File: iOS/Emotions/Emotions/Data/Auth/AuthAPI.swift
import Foundation

enum AuthAPI {
    case signIn(request: SignInRequestDTO)
    case signUp(request: SignUpRequestDTO)
}

// MARK: - Endpoint
extension AuthAPI: Endpoint {
    var path: String {
        switch self {
        case .signIn:
            return "auth/sign-in"
        case .signUp:
            return "auth/sign-up"
        }
    }

    var method: HTTPMethod {
        switch self {
        case .signIn, .signUp:
            return .post
        }
    }

    var body: Encodable? {
        switch self {
        case .signIn(let request):
            return request
        case .signUp(let request):
            return request
        }
    }

    var requiresAuth: Bool {
        false
    }
}

File: Android/app/src/main/java/com/example/emotionapp/data/api/AuthApi.kt
package com.example.emotionapp.data.api

import com.example.emotionapp.data.dtos.AuthResponseDTO
import com.example.emotionapp.data.dtos.SignInRequestDTO
import com.example.emotionapp.data.dtos.SignUpRequestDTO
import retrofit2.Response
import retrofit2.http.*

interface AuthApi {

    @POST("auth/sign-up")
    suspend fun signUp(
        @Body body: SignUpRequestDTO
    ): Response<AuthResponseDTO>

    @POST("auth/sign-in")
    suspend fun signIn(
        @Body body: SignInRequestDTO
    ): Response<AuthResponseDTO>
}